### ISSUE TITLE
Feature: provide a dashboard to configure OpenFaaS Cloud

### ISSUE THREAD

**Issue by matipan** [openfaas/openfaas-cloud#449]
There are a few things that are configured as environment variables in the different ofc components, a few of them are:
* Branch to be used for building the functions
* Docker registry to push images
* Custom templates
* Enable support for dockerfile templates
* Scaling factors
* Maybe hmac validation?

## Expected Behaviour

Be able to change the settings of OFC without requiring to modify each of the components that provide that feature. 

## Current Behaviour

If I want to modify, for example, the build branch I need to edit the deployments of many functions and this type of editing can be tedious and error prone.

## Possible Solution

Add a new component(maybe admin-dashboard?) to OFC that provides this features. I think we can do this in two different ways:
* Modify each of the required deployments directly from this component(needs the appropriate roles for modifying this) 
* Re-deploy the affected functions using the gateway(like buildshiprun does it). In order to do this we would need to first get the definition of the function, modify the variables and re-deploy with those new values.

In my opinion the second solution is best since the gateway continues to be the "owner" of the state of the functions.

**Comment by matipan** [comment]
Derek add label: area/ux,enhancement,help wanted

**Comment by alexellis** [comment]
Hmm you do seem to have Derek access https://github.com/openfaas/faas/blob/master/.DEREK.yml#L17

**Comment by matipan** [comment]
Yeah, I don't know why it didn't work. I think the syntax of the comment is ok

**Comment by alexellis** [comment]
```
matipan wants to add label of 'area/ux, enhancement, help wanted' on issue #449 
POST https://api.github.com/repos/openfaas/openfaas-cloud/issues/449/labels: 422 Validation Failed [{Resource:Label Field:name Code:invalid Message:}]
```

Add them separately

**Comment by matipan** [comment]
Derek add label: area/ux

**Comment by matipan** [comment]
Derek add label: enhancement

**Comment by matipan** [comment]
Derek add label: help wanted

**Comment by matipan** [comment]
There we go. Thanks @alexellis
